js/ai/src/model.ts

```diff
@@ -84,19 +108,39 @@ export const ToolResponsePartSchema = EmptyPartSchema.extend({
     output: z.unknown().optional(),
   }),
 });
+
+/**
+ * Tool response part.
+ */
 export type ToolResponsePart = z.infer<typeof ToolResponsePartSchema>;
 
+/**
+ * Zod schema of a data part.
+ */
 export const DataPartSchema = EmptyPartSchema.extend({
   data: z.unknown(),
 });
 
+/**
+ * Data part.
+ */
 export type DataPart = z.infer<typeof DataPartSchema>;
 
+/**
+ * Zod schema of a custom part.
+ */
 export const CustomPartSchema = EmptyPartSchema.extend({
   custom: z.record(z.any()),
```

Comment:
nit: what is custom part?

Reply:
It's used for custom model responses, e.g. Gemini code execution.